pkg/vm/stack_item.go

```diff
@@ -481,24 +481,24 @@ func (i *BoolItem) Convert(typ StackItemType) (StackItem, error) {
 
 // ByteArrayItem represents a byte array on the stack.
 type ByteArrayItem struct {
-	value []byte
+	value string
 }
 
 // NewByteArrayItem returns an new ByteArrayItem object.
 func NewByteArrayItem(b []byte) *ByteArrayItem {
 	return &ByteArrayItem{
-		value: b,
+		value: string(b),
 	}
 }
 
 // Value implements StackItem interface.
 func (i *ByteArrayItem) Value() interface{} {
-	return i.value
+	return []byte(i.value)
```

Comment:
I consider `string` an internal detail of `ByteArray` implementation, because it is not really a string more like "immutable slice of bytes". It is more convenient to be able to use all `Buffer`/`ByteArray` the same way (e.g. in `popSigElements`).

Reply:
I think if we're not changing externally-visible type than this change doesn't make much sense as it adds some conversions where they're better be avoided.

Reply:
FIxed.